### PR TITLE
FreeBSD/clang update proposal.

### PR DESCRIPTION
**Pull request by devnexen**
silence few build warnings, VKDMA: disable Intel chipset for host buffer as it requires root privilege.

### CODE REVIEW

File: rpcs3/Emu/RSX/VK/VKDMA.cpp [hunk 1]
```diff
@@ -279,7 +279,13 @@ namespace vk
 		}
 #else
 		// Anything running on AMDGPU kernel driver will not work due to the check for fd-backed memory allocations
-		const bool allow_host_buffers = (vendor != driver_vendor::AMD && vendor != driver_vendor::RADV);
+		const bool allow_host_buffers = (
```
Comment (Megamouse) [inline rpcs3/Emu/RSX/VK/VKDMA.cpp:282]: I think this looks much better
```suggestion
#if defined(__FreeBSD__)
		// Requires root privilege on FreeBSD on Intel
		const bool allow_host_buffers = (vendor != driver_vendor::AMD && vendor != driver_vendor::RADV);
#else
		const bool allow_host_buffers = (vendor != driver_vendor::AMD && vendor != driver_vendor::RADV && vendor != driver_vendor::INTEL);
#endif
```

Comment (devnexen) [reply rpcs3/Emu/RSX/VK/VKDMA.cpp:282]: Yes just tried to avoid the repetition but ok no pb.

File: rpcs3/Emu/RSX/VK/VKDMA.cpp [hunk 2]
```diff
@@ -279,7 +279,14 @@ namespace vk
 		}
 #else
```
Comment (Megamouse) [inline rpcs3/Emu/RSX/VK/VKDMA.cpp:280]: Ups, I missed this. Actually you can transform this into an #else if defined(__FreeBSD__)

File: rpcs3/Emu/RSX/VK/VKDMA.cpp [hunk 3]
```diff
@@ -277,6 +277,10 @@ namespace vk
 		{
 			allow_host_buffers = true;
 		}
+#elif defined(__FreeBSD__)
```
Comment (jbeich) [inline rpcs3/Emu/RSX/VK/VKDMA.cpp:280]: Looks similar to https://github.com/freebsd/freebsd-ports/commit/b8c1cb4ca5269441f4ba7d3828268f9099f702a9. However, gating on `__FreeBSD__` doesn't make sense. DRM_IOCTL_I915_GEM_USERPTR isn't implemented on any other BSD.

File: rpcs3/Emu/RSX/VK/VKDMA.cpp [hunk 4]
```diff
@@ -277,6 +277,10 @@ namespace vk
...
+		// Anything running on AMDGPU kernel driver will not work due to the check for fd-backed memory allocations
+		// Requires root privilege on FreeBSD on Intel
```
Comment (jbeich) [inline rpcs3/Emu/RSX/VK/VKDMA.cpp:282]: root is specific to `mesa-devel` as `mesa-dri` package lacks https://github.com/freebsd/freebsd-ports/commit/058e5ba0ff29ae64383b8392f23facad616be8c9. With vanilla Mesa VK_EXT_external_memory_host simply fails unless user hides the extension from RPCS3 e.g., via a Vulkan layer.

File: rpcs3/Emu/RSX/VK/VKDMA.cpp (hunk 3 above)
Comment (devnexen) [reply rpcs3/Emu/RSX/VK/VKDMA.cpp:280]: Ah I did not know about other BSDs, FreeBSD was the only one I tried.

**Review by kd-11 — APPROVED** [review]
This is fine until automatic detection of allocation failure is implemented

**Pull request merged**